examples/sfu-ws/main.go

```diff
@@ -0,0 +1,216 @@
+							fmt.Println(err)
+						}
+					}
+				}()
+
+				rtpBuf := make([]byte, 1400)
+				for {
+					i, err := remoteTrack.Read(rtpBuf)
+					checkError(err)
+					_, err = videoTrack.Write(rtpBuf[:i])
+					checkError(err)
+				}
+
+			} else {
+
+				// Create a local audio track, all our SFU clients will be fed via this track
+				var err error
+				audioTrack, err = PubReceiver.NewTrack(remoteTrack.PayloadType(), remoteTrack.SSRC(), "audio", "pion")
+				checkError(err)
+
+				rtpBuf := make([]byte, 1400)
+				for {
+					i, err := remoteTrack.Read(rtpBuf)
+					checkError(err)
+					_, err = audioTrack.Write(rtpBuf[:i])
+					checkError(err)
+				}
+			}
+		})
+
+		offer := webrtc.SessionDescription{
+			SDP:  string(msg),
+			Type: webrtc.SDPTypeOffer,
+		}
+
+		// Set the remote SessionDescription
+		checkError(PubReceiver.SetRemoteDescription(offer))
+
+		// Create answer
+		answer, err := PubReceiver.CreateAnswer(nil)
+		checkError(err)
+
+		// Sets the LocalDescription, and starts our UDP listeners
+		checkError(PubReceiver.SetLocalDescription(answer))
+
+		// Send server sdp to publisher
+		checkError(c.WriteMessage(mt, []byte(answer.SDP)))
+		fmt.Println("send pion sdp to publisher:==============\n" + answer.SDP)
+	} else {
+		fmt.Println("scriber sdp:=============\n", string(msg))
+
+		recvOnlyOffer := webrtc.SessionDescription{
+			SDP:  string(msg),
+			Type: webrtc.SDPTypeOffer,
+		}
+
+		// Create a new PeerConnection
+		SubSender, err := api.NewPeerConnection(peerConnectionConfig)
+		checkError(err)
+
+		// Waiting for publisher track finish
+		for {
+			if videoTrack == nil {
+				fmt.Println("videoTrack == nil, waiting..")
+				time.Sleep(100 * time.Millisecond)
+			} else {
+				break
+			}
+		}
+
+		// Add local video track
+		_, err = SubSender.AddTrack(videoTrack)
+		checkError(err)
+
+		// Add local audio track
+		_, err = SubSender.AddTrack(audioTrack)
+		checkError(err)
+
+		// Set the remote SessionDescription
+		checkError(SubSender.SetRemoteDescription(recvOnlyOffer))
+
+		// Create answer
+		answer, err := SubSender.CreateAnswer(nil)
+		checkError(err)
+
+		// Sets the LocalDescription, and starts our UDP listeners
+		checkError(SubSender.SetLocalDescription(answer))
+
+		// Send server sdp to subscriber
+		checkError(c.WriteMessage(mt, []byte(answer.SDP)))
+		fmt.Println("send pion sdp to subscriber:==============\n" + answer.SDP)
+	}
+}
+
+func web(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
+		t, _ := template.ParseFiles("sfu.html")
+		checkError(t.Execute(w, nil))
+	}
+}
+
+func main() {
+
+	// Enable pprof
+	go func() {
+		fmt.Println("pprof listen 6060")
+		http.ListenAndServe(":6060", nil)
+	}()
+
+	// Websocket handle func
+	http.HandleFunc("/ws", Room)
+
+	// Html handle func
+	http.HandleFunc("/", web)
+
+	// Support https, so we can test by lan
+	if err := http.ListenAndServeTLS(":443", "cert.pem", "key.pem", nil); err != nil {
```

Comment:
Is it possible to generate the cert/key in the example? I don't want to encourage the reuse (or committing secrets in git)

Reply:
Can you just do `panic(http.ListenAndServerTLS())` that function should never return (unless we have an error)

Reply:
Can you make the port configurable, I never run others code as `root` just to be safe.

Reply:
Good idea, i will fix.